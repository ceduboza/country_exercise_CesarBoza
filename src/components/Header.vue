<template>
    <div class="header">
        <h1>{{ title }}</h1>
        <slot />
    </div>
</template>
<script setup lang="ts">
import { defineProps } from 'vue';

defineProps({
    title: String
})

</script>
<style lang="scss">
.header {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 5vh 5vw;
}
</style>