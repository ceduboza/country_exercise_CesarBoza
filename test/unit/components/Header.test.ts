import { describe, it, expect } from "vitest";
import { shallowMount } from "@vue/test-utils";
import Header from "../../../src/components/Header.vue";

describe("Header", () => {
    let wrapper

    const findTitleHeader = wrapper => wrapper.find("h1")
    const findSlotContent = wrapper => wrapper.find("div.slot-content")

    const factory = ({ props = {}, slots = {} } = {}) => {
        wrapper = shallowMount(Header, {
            props,
            slots
        })
    }

    it("should render the header", () => {
        const props = {
            title: "Header",
        }
        const slots = {
            default: `<div class="slot-content">Slot</div>`
        }

        factory({ props, slots })

        const titleHeader = findTitleHeader(wrapper)

        expect(wrapper.html()).toMatchSnapshot()
        expect(titleHeader.exists()).toBe(true)
        expect(titleHeader.text()).toBe("Header")
        expect(findSlotContent(wrapper).exists()).toBe(true)
    })
});
