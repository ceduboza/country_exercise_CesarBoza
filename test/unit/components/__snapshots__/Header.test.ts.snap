// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should render the header 1`] = `
"<div class="header">
  <h1>Header</h1>
  <div class="slot-content">Slot</div>
</div>"
`;
